# Individual Project Planning Assignment

## 1) Data Description

In [1]:
library(repr)
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [3]:
players_data <- read_csv("https://drive.google.com/uc?export=download&id=1Mw9vW0hjTJwRWx0bDXiSpYsO3gKogaPz")
players_data

Rows: 196 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, age
lgl (3): subscribe, individualId, organizationName

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1   Pro         TRUE    
2   Veteran     TRUE    
3   Veteran    FALSE    
4   Amateur     TRUE    
5   Regular     TRUE    
6   Amateur     TRUE    
7   Regular     TRUE    
8   Amateur    FALSE    
9   Amateur     TRUE    
10  Veteran     TRUE    
11  Veteran     TRUE    
12  Pro         TRUE    
13  Beginner    TRUE    
14  Amateur     TRUE    
15  Veteran     TRUE    
16  Amateur    FALSE    
17  Beginner    TRUE    
18  Amateur     TRUE    
19  Amateur     TRUE    
20  Regular     TRUE    
21  Regular     TRUE    
22  Amateur    FALSE    
23  Beginner    TRUE    
24  Beginner    TRUE    
25  Amateur     TRUE    
26  Regular     TRUE    
27  Regular     TRUE    
28  Veteran    FALSE    
29  Amateur     TRUE    
30  Veteran    FALSE    
⋮   ⋮          ⋮        
167 Amateur    FALSE    
168 Beginner   FALSE    
169 Veteran     TRUE    
170 Veteran    FALSE    
171 Pro        FALSE    
172 Beginner   FALSE    
173 Veteran     TRUE    
174 Regular    FALSE    
175 Veteran     TRUE    
176 Amateur     TRUE    
177 Pro        FALSE    
178 Veteran    FALSE    
179 Amateur     TRUE    
180 Veteran     TRUE    
181 Amateur    FALSE    
182 Amateur     TRUE    
183 Pro        FALSE    
184 Amateur     TRUE    
185 Pro        FALSE    
186 Regular    FALSE    
187 Veteran     TRUE    
188 Amateur     TRUE    
189 Beginner    TRUE    
190 Amateur    FALSE    
191 Amateur     TRUE    
192 Amateur     TRUE    
193 Veteran    FALSE    
194 Amateur    FALSE    
195 Amateur    FALSE    
196 Pro         TRUE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3   b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
4   23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
5   7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e
6   f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977
7   8e594b8953193b26f498db95a508b03c6fe1c24bb5251d392c18a0da9a722807
8   1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd
9   8b71f4d66a38389b7528bb38ba6eb71157733df7d1740371852a797ae97d82d1
10  bbe2d83de678f519c4b3daa7265e683b4fe2d814077f9094afd11d8f217039ec
11  7a4686586d290c67179275c7c3dfb4ea02f4d317d9ee0e2cee98baa27877a875
12  4caa42e1b20511552434978171dcf7283fb6eb857eb871327e39bfc328002309
13  4bed6f49b672974f706aa7f020b4f29020326bf594520593557cd381a51559b8
14  3caa832978e0596779f4ee7c686c4592fb6de893925025d25bf26be9a8cd03f5
15  6f9acf8ea9956fe817895c78d10e1e25c11aba335a451e5b0322b073ee5008c4
16  2313a06afe47eacc28ff55adf6f072e7d12b0d12d7cbae29656f9029ec72bb53
17  9db37a1d381697f6c83e3b97b276e38bb9d9c811ba4596edb75d4dbfea6415b0
18  a175d4741dc84e6baf77901f6e8e0a06f54809a34e6b5211159bced346f7fb3e
19  ab1f44f93c3b828f55458971db393052d9711df3e0e7ff69540bfebfcec555ff
20  20ae3f8b5323674fe6e537bc91928841cfc2a00c1b36522fd25370bb6fa89d36
21  bc704ff2bc676dbf48ee41b9e11481c1387bf758ad318f2428f336e3fecc6660
22  200db0417ee70863c7e2a0f214c737476fb86de5869d0b907a077d726357e687
23  f9ac013b2f0bc2bd4928a6a0fc8a0aae8b5c4f2670cf9e18f547b7e72034c965
24  89c8e07d6a2aa102d9cd05086d51b2e41195c9c807dd976a06f4262ff36d0cca
25  119f01b9877fc5ea0073d05602a353b91c4b48e4cf02f42bb8d661b46a34b760
26  5baba1651a0b92788bc0d6dcdf00be64af1cf9f0015bbeed204ddabd1d872e83
27  50cec73a632e4be7b155c3df5165de28b2c5baf092abcb7f509f5645776a56c2
28  f8acd26a5e452b682b9f8b20108caef1ee2e745efe08e95806980bd2388a3fea
29  4b01bce3f141289709e8278b02ba5d2aaa7105d7ccb9c7deb37670a80e332774
30  951e54f7376e2b2f0915e9e3646c701af4a2fe839385b13f9e0084b08f8070bd
⋮   ⋮                                                               
167 7096b30972a32029244fc6171f21bc5996b51b8d75df7a94f003dead428988f9
168 42eafe96ed5c1684e3b5cc614d1b01a117173d3ec6898abcf1177d9822bd39f7
169 9ba67e127e4e70d909f6237644bb4674b17bc6ae5c9e735f88cfc50e22b499c6
170 4bfad3613c71ace05644bf210195d9fb0d3

By looking at the data sets above of the players and sessions, we can see that for the players data set, there are 196 observations and 7 variables (experience level, whether the player is a subscriber, their hashed email, number of hours played, the player’s name, their gender, and their age), excluding the columns individualId and organizationName which only contain NA. The data is tidy as each column corresponds to a single variable, each observation forms a row, and each cell contains a single value.

A potential issue with the data is that some of the values in the columns may contain NA, which would cause problems when wrangling and analyzing. Unseen issues could include bias from how the data was collected, since there is only information on players who voluntarily joined the survey and not on MineCraft players as a whole. Furthermore, for the players' age and experience, we do not know how this information was obtained - if it was self-reported, we do not know how reliable that information is.

#### Summary Statistics

In [5]:
summary(players_data)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
     name              gender               age        individualId  
 Length:196         Length:196         Min.   : 8.00   Mode:logical  
 Class :character   Class :character   1st Qu.:17.00   NA's:196      
 Mode  :character   Mode  :character   Median :19.00                 
                                       Mean   :21.28                 
                                       3rd Qu.:22.00                 
                                       Max.   :99.00                 

In [6]:
player_genders <- players_data |>
    group_by(gender) |>
    summarize(count = n())
player_experience <- players_data |>
    group_by(experience) |>
    summarize(count = n())
player_subscribers <- players_data |>
    group_by(subscribe) |>
    summarize(count = n())
player_genders
player_experience
player_subscribers

gender            count
1 Agender             2  
2 Female             37  
3 Male              124  
4 Non-binary         15  
5 Other               1  
6 Prefer not to say  11  
7 Two-Spirited        6

experience count
1 Amateur    63   
2 Beginner   35   
3 Pro        14   
4 Regular    36   
5 Veteran    48

subscribe count
1 FALSE      52  
2  TRUE     144

By observing the statistics above, we can see that for this data set, there are more subscribers than non-subscribers, most players are amateur level, and most are male. We also know that the max age of the players is 99 while the mean is 21.28 years old. The max number of hours played is 223.10 while the mean is 5.85 hours.

Another issue that can be identified from the statistics above is that if we were to use the data for a classification problem, some of the variables in this dataset exhibit oversampling (e.g. more data on male players than other gender players), which would lower the accuracy of a KNN classification model.

In [11]:
experience_summary <- players_data |>
    group_by(experience) |>
    summarise(avg_played_hrs = mean(played_hours, na.rm = TRUE), max_hrs_played = max(played_hours, na.rm = TRUE),
              avg_age = mean(age, na.rm = TRUE), max_age = max(age, na.rm = TRUE))

subscriber_summary <- players_data |>
    group_by(subscribe) |>
    summarise(avg_played_hrs = mean(played_hours, na.rm = TRUE), max_hrs_played = max(played_hours, na.rm = TRUE),
              avg_age = mean(age, na.rm = TRUE), max_age = max(age, na.rm = TRUE))

gender_summary <- players_data |>
    group_by(gender) |>
    summarise(avg_played_hrs = mean(played_hours, na.rm = TRUE), max_hrs_played = max(played_hours, na.rm = TRUE),
              avg_age = mean(age, na.rm = TRUE), max_age = max(age, na.rm = TRUE))

experience_summary
subscriber_summary
gender_summary

experience avg_played_hrs max_hrs_played avg_age  max_age
1 Amateur     6.0174603     150.0          20.25397 45     
2 Beginner    1.2485714      23.7          21.65714 42     
3 Pro         2.6000000      30.3          22.21429 91     
4 Regular    18.2083333     223.1          22.77778 99     
5 Veteran     0.6479167      12.5          20.95833 46

subscribe avg_played_hrs max_hrs_played avg_age  max_age
1 FALSE     0.5019231        7.1          22.98077 50     
2  TRUE     7.7756944      223.1          20.66667 99

gender            avg_played_hrs max_hrs_played avg_age  max_age
1 Agender            6.25000000     12.5          23.50000 27     
2 Female            10.63513514    178.2          21.81081 44     
3 Male               4.12741935    223.1          20.20968 50     
4 Non-binary        14.88000000    218.1          19.06667 25     
5 Other              0.20000000      0.2          91.00000 91     
6 Prefer not to say  0.37272727      2.0          21.36364 33     
7 Two-Spirited       0.08333333      0.4          33.16667 99

| Player Status | Average Number of Hours Played | Highest Number of Hours Played | Average Age | Highest Age |
|----------------|--------------------------------|--------------------------------|-------------| ------------|
| Subscribed | 7.78 | 223.1 | 20.67 | 99 |
| Not Subscribed | 0.5 | 7.1 | 22.98 | 50 |
| Amateur | 6.02 | 150 | 20.25 | 45 |
| Beginner | 1.25 | 23.7 | 21.66 | 42 |
| Pro | 2.6 | 30.3 | 22.21 | 91 |
| Regular | 18.21 | 223.1 | 22.78 | 99 |
| Veteran | 0.65 | 12.5 | 20.96 | 46 |


| Player Gender | Average Number of Hours Played | Highest Number of Hours Played | Average Age | Highest Age |
|-------------------|--------------------------------|--------------------------------|-------------|-------------|
| Agender | 6.25 | 12.5 | 23.5 | 27 |
| Female | 10.64 | 178.2 | 21.81 | 44 |
| Male | 4.13 | 223.1 | 20.21 | 50 |
| Non-binary | 14.88 | 218.1 | 19.07 | 25 |
| Other | 0.2 | 0.2 | 91 | 91 |
| Prefer Not To Say | 0.37 | 2 | 21.36 | 33 |
| Two-Spirited | 0.08 | 0.4 | 33.17 | 99 |

From the tables of summary statistics above, we can see that subscribers and regular level players play the most, while non-subscribers and veteran level players play the least. The average age appears to be about the same across all types of players (20 - 23), so age does not appear to change/differ with a player's subscription status and experience. We can also see that male players have the highest number of hours played while non-binary players have the highest average played hours. Average age fluctuates a bit more for player gender, ranging from 20 - 91.

### 2) Questions

The question I selected for this project was Question 2 below.

**Broad Question (Q2):** What “kinds” of players are most likely to contribute a large amount of data?

**Specific (Data Analysis) Question:** Can the experience level, subscription status, and age of a player predict the number of hours of MineCraft they have played in the players data set?

The data will help me address my question of interest as I can use the experience, subscribe, and age variables in the data set to predict on the played_hours variable. Because the players data is already tidy, there is no need to wrangle it before the analysis.

### 3) Exploratory Data Analysis and Visualization